test/utils.test.js

```diff
@@ -14,37 +38,367 @@ describe('utils', function() {
+      assert.doesNotThrow(() => utils.appendChild(['p'], child));
+    });
+
+    describe('when appending a fragment', function() {
+      it('should append all elements', function() {
+        const jml = ['div'];
+        const fragment = ['',
+          ['em', 'hello'],
+          ['strong', 'world'],
+        ];
+        utils.appendChild(jml, fragment);
+        assert.deepEqual(jml, ['div', ['em', 'hello'], ['strong', 'world']]);
+      });
+
+      // @TODO - is this an intentional mutation?
+      it('should remove fragment contents', function() {
+        const jml = ['div'];
+        const fragment = ['',
+          ['em', 'hello'],
+          ['strong', 'world'],
+        ];
+        utils.appendChild(jml, fragment);
+        assert.deepEqual(fragment, []);
+      });
+    });
+
+    describe('when appending an element', function() {
+      it('should throw an error if child is not a valid element', function() {
+        assert.throws(() => utils.appendChild(['div'], []), SyntaxError);
+        assert.throws(() => utils.appendChild(['div'], [5]), SyntaxError);
+        assert.throws(() => utils.appendChild(['div'], [null]), SyntaxError);
+        assert.throws(() => utils.appendChild(['div'], [true]), SyntaxError);
+      });
+
+      it('should append a valid child element', function() {
+        const jml = ['div'];
+        const child = ['p'];
+        utils.appendChild(jml, child);
+        assert.deepEqual(jml, ['div', child]);
+      });
+
+      it('should append element with all descendents', function() {
+        const jml = ['div', { a: 1}];
+        const subtree = ['section',
+          ['p', 'hello'],
+          ['p', 'world'],
+        ];
+        utils.appendChild(jml, subtree);
+        assert.deepEqual(jml, ['div', { a: 1 }, subtree]);
+      });
+
+      it('should execute an optional callback before appending (for JBST use)', function(done) {
+        const jml = ['div'];
+        const el = ['p'];
+        utils.appendChild(jml, el, function onAppend(parent, child) {
+          assert.strictEqual(jml, parent);
+          assert.strictEqual(el, child);
+          assert.deepEqual(jml, ['div']);
+          done();
+        });
+      });
+    });
+
+    describe('when appending a raw child', function() {
+      // @TODO - this will currently throw an error due to missing function `isRaw`
+      xit('should append to the parent', function() {
+        const jml = ['div'];
+        const child = new function RawChild() {};
+        utils.appendChild(jml, child);
+        assert.strictEqual(jml, ['div', child]);
+      });
+    });
+
+    describe('when appending attributes', function() {
+      // @TODO - this will currently throw an error due to missing function `isRaw`
+      xit('should add attributes to the parent element', function() {
+        const jml = ['div'];
+        const attrs = { a: 1 };
+        utils.appendChild(jml, attrs);
+        assert.deepEqual(jml, ['div', { a: 1 }]);
+      });
```

Comment:
This also fails but it is due to the error above. The `isRaw` call happens before the method can reach this logic.

Reply:
Just raise a new issue, I need to to read the source code, and may be re-design these APIs.

Reply:
Also fixed by #10